server/src/main/java/org/opensearch/http/AbstractHttpServerTransport.java

```diff
@@ -438,29 +448,56 @@ private void handleIncomingRequest(final HttpRequest httpRequest, final HttpChan
             RestChannel innerChannel;
             ThreadContext threadContext = threadPool.getThreadContext();
             try {
-                innerChannel = new DefaultRestChannel(
-                    httpChannel,
-                    httpRequest,
-                    restRequest,
-                    bigArrays,
-                    handlingSettings,
-                    threadContext,
-                    corsHandler,
-                    trace
-                );
+                if (httpChannel instanceof StreamingHttpChannel) {
```

Comment:
Not particularly proud of `instanceof` usage but the reason for it is to evolve APIs in a non-breaking ways so we could backport this change to `2.x` line easily.

Reply:
I don't have an alternative suggestion for the `instanceof`, but is this identical to the `if` condition on line 478? If so, refactoring out to a private method might be an option.

Reply:
No, it is not identical from arguments perspective, I could introduce a method surely but I don't to much benefit to be fair, this is very self contained code in a scope of one method.